pyrate/core/dem_error.py

```diff
@@ -0,0 +1,307 @@
+(i.e. local incidence and azimuth angles).
+"""
+# pylint: disable=invalid-name, too-many-locals, too-many-arguments
+import os
+import numpy as np
+from math import sqrt, sin, cos, tan, asin, atan, atan2, isnan, pi
+from typing import Optional, List, Dict, Iterable
+
+from pyrate.core import shared, ifgconstants as ifc, mpiops, config as cf
+from pyrate.core.logger import pyratelogger as log
+from pyrate.core.shared import Ifg
+from pyrate.core.refpixel import convert_pixel_value_to_geographic_coordinate
+from pyrate.core.shared import read_lookup_table
+from pyrate.configuration import MultiplePaths
+
+
+def remove_dem_error(ifgs: List, params: dict) -> None:
+    """
+    Wrapper function for PyRate DEM error removal functionality.
+    """
+    # read first IFG
+    ifg_paths = [ifg_path.tmp_sampled_path for ifg_path in params[cf.INTERFEROGRAM_FILES]]
+    ifg0_path = ifg_paths[0]
+    ifg0 = Ifg(ifg0_path)
+    ifg0.open(readonly=True)
+
+    # calculate per-pixel lon/lat
+    lon, lat = get_lonlat_coords(ifg0)
+    # calculate per-pixel radar coordinates
+    az, rg = get_radar_coords(ifg0, params)
+    # calculate per-pixel look angle (also calculates and saves indcidence and azimuth angles)
+    look_angle = calc_local_geometry(ifg0, ifg0_path, az, rg, lon, lat, params)
+    # loop over all Ifgs in stack
+    ifgs = [shared.Ifg(p) for p in ifg_paths] if isinstance(ifgs[0], str) else ifgs
+    process_ifgs = mpiops.array_split(ifgs)
+    for ifg in process_ifgs:
+        bperp = calc_local_baseline(ifg, az, look_angle, params)
+    # todo: add code to invert for DEM error using the per-pixel baseline values and the unwrapped phase
+
+
+def get_lonlat_coords(ifg):
+    """
+    Function to get longitude and latitude coordinates for each pixel in the multi-looked.
+    interferogram dataset. Coordinates are identical for each interferogram in the stack.
+    """
+    # assume all interferograms have same projection and will share the same transform
+    transform = ifg.dataset.GetGeoTransform()
+    # number of rows and columns in dataset
+    nrows, ncols = ifg.shape
+    lon = np.zeros((nrows, ncols))  # pre-allocate 2D numpy array
+    lat = np.zeros((nrows, ncols))  # pre-allocate 2D numpy array
+    for i in range(0, nrows): # rows are y-direction
+       for j in range(0, ncols): # cols are x-direction
+           lon_ij, lat_ij = convert_pixel_value_to_geographic_coordinate(j, i, transform)
+           lon[i, j] = lon_ij
+           lat[i, j] = lat_ij
+
+    return lon, lat
+
+
+def get_radar_coords(ifg, params):
+    """
+    Function to get radar coordinates for each pixel in the multi-looked interferogram dataset.
+    Radar coordinates are identical for each interferogram in the stack.
+    """
+    # lookup table file:
+    lookup_table = params[cf.LT_FILE]
+    # PyRate IFG multi-looking factors
+    ifglksx = params[cf.IFG_LKSX]
+    ifglksy = params[cf.IFG_LKSY]
+    print(lookup_table)
```

Comment:
Is this just for testing? Use a `log.debug` statement instead of print if you want it to persevere in the code

Reply:
Was just for testing. Deleted.